doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -92,6 +92,155 @@ writeShellScript "evaluate-my-file.sh" ''
 ```
 ::::
 
+### `makeDesktopItem` {#trivial-builder-makeDesktopItem}
+
+Write an [XDG desktop file](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) to the Nix store.
+
+This function is most often used with `mkDerivation.desktopItems` to add desktop items to a package.
+
+`makeDesktopItem` takes an attribute set according to the [XDG specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys), with the following possible values:
```

Comment:
```suggestion
#### Inputs {#trivial-builder-makeDesktopItem-inputs}

`makeDesktopItem` takes an attribute set according to the [XDG specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys), with the following possible values:
```

Also, if all fields from the specification link are currently satisfied by `makeDesktopItem`, I wouldn't bother listing all of them below as you did, but would rather put on a single line.
It looks like `makeDesktopItem`'s attributes do not correspond 1:1 with the specification, so you have to point out that additional attributes are also supported.

Something like:

`makeDesktopItem` takes an attribute set that also accepts all values from the [XDG specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys).

All recognised keys from the specification are supported with the exception of the "Hidden" field. The keys are converted into camelCase format, but correspond 1:1 to their equivalent in the specification: `genericName`, `noDisplay`, `comment`, (list all of them here).

The "Version" field is hardcoded... (see another comment about this)

The following fields are either required, carry specific default values, or are additional fields supported by `makeDesktopItem`:

(info about `name`, `type`, `desktopName`, `extraConfig`, ...)

Reply:
How should we list the type for each field in that case?

Reply:
I didn't look at each attribute and compare with the XDG specification, but I was assuming the types would map 1:1 to whatever is in the specification.

For example, see the documentation for the `config` attribute for [`dockerTools.buildImage`'s inputs](https://nixos.org/manual/nixpkgs/unstable/#ssec-pkgs-dockerTools-buildImage-inputs). It does something similar just pointing to the spec.

If there's only one or two extra attributes with different types compared to the spec, let's document those properly in our docs, but still point to the spec for the other ones. If most of the attributes are different types, then I take back most of what I said, and I think it's preferable for us to document all of them on our side.